### PR TITLE
fix new validation errors due to ADV/SCONJ changes in #90

### PR DESCRIPTION
**Pull request by nschneid**
This is to fix issues that must be due to edge cases or errors in the original data not addressed by the DepEdit rules in #90. In particular, the `leaf-mark-case` errors occur in cases where "why/when/whenever/whatever" was analyzed as having a relative clause modifier and has now been changed to SCONJ.

Two issues of note:

weblog-juancole.com_juancole_20040823064025_ENG_20040823_064025-0086
"Mr. Bush would also sometimes tell stories about his days at Yale in New Haven, and **how whenever** he got pulled over for erratic driving, he was let go"

[his days at Yale in New Haven] is coordinated with [how [whenever he got pulled over for erratic driving], he was let go].
The validator is triggered by the relative clause analysis of "whenever + he got pulled over", which I will fix to match the SCONJ policy. But I am still not sure about the "how": should it remain conj(stories, let), mark(let, how/SCONJ)?

reviews-063549-0002: "Theres a reason why Frank mcclelland was named best chef of the north east reigon."

I can fix the validator error here, which is caused by the attachment of "why", but I am not sure of the relation between "reason" and "named", which is currently `acl:relcl`. In general, is the relative clause analysis appropriate for clauses licensed by "reason"? Searching the corpus for "reason" I see it is the typical analysis in examples such as "the main reason I wrote this post", "the primary reason we need these things", "the reason Senators Daschle and Leahy would have been targeted", "the only reason I would go to this particular QT", "the reason he transferred to the Alabama Air National Guard". But I wonder if it should be considered `acl` instead, because "reason" is not extracted from a core position of the embedded clause.

### CODE REVIEW

**Comment by sebschu** [comment]
_whenever he got pulled over_ is a free relative (see https://nlp.stanford.edu/pubs/demarneffe13more.pdf), so I think _whenever_ should be `ADV` and attached to _let-36_ as an advmod.

And I think `conj(stories, let)`, `mark(let, how/SCONJ)` is the correct analysis.

```
# sent_id = weblog-juancole.com_juancole_20040823064025_ENG_20040823_064025-0086
# text = GOODWYN: According to Archibald, Mr. Bush would also sometimes tell stories about his days at Yale in New Haven, and how whenever he got pulled over for erratic driving, he was let go after the officers discovered he was the grandson of a Connecticut US senator.
1	GOODWYN	GOODWYN	PROPN	NNP	Number=Sing	0	root	0:root	SpaceAfter=No
2	:	:	PUNCT	:	_	1	punct	1:punct	_
3	According	accord	VERB	VBG	VerbForm=Ger	5	case	5:case	_
4	to	to	ADP	IN	_	5	case	5:case	_
5	Archibald	Archibald	PROPN	NNP	Number=Sing	12	obl	12:obl:to	SpaceAfter=No
6	,	,	PUNCT	,	_	12	punct	12:punct	_
7	Mr.	Mr.	PROPN	NNP	Number=Sing	12	nsubj	12:nsubj	_
8	Bush	Bush	PROPN	NNP	Number=Sing	7	flat	7:flat	_
9	would	would	AUX	MD	VerbForm=Fin	12	aux	12:aux	_
10	also	also	ADV	RB	_	12	advmod	12:advmod	_
11	sometimes	sometimes	ADV	RB	_	12	advmod	12:advmod	_
12	tell	tell	VERB	VB	VerbForm=Inf	1	parataxis	1:parataxis	_
13	stories	story	NOUN	NNS	Number=Plur	12	obj	12:obj	_
14	about	about	ADP	IN	_	16	case	16:case	_
15	his	he	PRON	PRP$	Gender=Masc|Number=Sing|Person=3|Poss=Yes|PronType=Prs	16	nmod:poss	16:nmod:poss	_
16	days	day	NOUN	NNS	Number=Plur	13	nmod	13:nmod:about	_
17	at	at	ADP	IN	_	18	case	18:case	_
18	Yale	Yale	PROPN	NNP	Number=Sing	16	nmod	16:nmod:at	_
19	in	in	ADP	IN	_	21	case	21:case	_
20	New	New	PROPN	NNP	Number=Sing	21	compound	21:compound	_
21	Haven	Haven	PROPN	NNP	Number=Sing	16	nmod	16:nmod:in	SpaceAfter=No
22	,	,	PUNCT	,	_	36	punct	36:punct	_
23	and	and	CCONJ	CC	_	36	cc	36:cc	_
24	how	how	SCONJ	WRB	PronType=Int	36	mark	36:mark	_
25	whenever	whenever	ADV	WRB	PronType=Int	36	advmod	36:advmod	_
26	he	he	PRON	PRP	Case=Nom|Gender=Masc|Number=Sing|Person=3|PronType=Prs	28	nsubj:pass	28:nsubj:pass	_
27	got	get	AUX	VBD	Mood=Ind|Tense=Past|VerbForm=Fin	28	aux:pass	28:aux:pass	_
28	pulled	pull	VERB	VBN	Tense=Past|VerbForm=Part|Voice=Pass	25	acl:relcl	25:acl:relcl	_
29	over	over	ADV	RB	_	28	advmod	28:advmod	_
30	for	for	ADP	IN	_	32	case	32:case	_
31	erratic	erratic	ADJ	JJ	Degree=Pos	32	amod	32:amod	_
32	driving	driving	NOUN	NN	Number=Sing	28	obl	28:obl:for	SpaceAfter=No
33	,	,	PUNCT	,	_	36	punct	36:punct	_
34	he	he	PRON	PRP	Case=Nom|Gender=Masc|Number=Sing|Person=3|PronType=Prs	36	nsubj:pass	36:nsubj:pass|37:nsubj:xsubj	_
35	was	be	AUX	VBD	Mood=Ind|Number=Sing|Person=3|Tense=Past|VerbForm=Fin	36	aux:pass	36:aux:pass	_
36	let	let	VERB	VBN	Tense=Past|VerbForm=Part|Voice=Pass	16	conj	13:nmod:about|16:conj:and	_
37	go	go	VERB	VB	VerbForm=Inf	36	xcomp	36:xcomp	_
38	after	after	SCONJ	IN	_	41	mark	41:mark	_
39	the	the	DET	DT	Definite=Def|PronType=Art	40	det	40:det	_
40	officers	officer	NOUN	NNS	Number=Plur	41	nsubj	41:nsubj	_
41	discovered	discover	VERB	VBD	Mood=Ind|Tense=Past|VerbForm=Fin	36	advcl	36:advcl:after	_
42	he	he	PRON	PRP	Case=Nom|Gender=Masc|Number=Sing|Person=3|PronType=Prs	45	nsubj	45:nsubj	_
43	was	be	AUX	VBD	Mood=Ind|Number=Sing|Person=3|Tense=Past|VerbForm=Fin	45	cop	45:cop	_
44	the	the	DET	DT	Definite=Def|PronType=Art	45	det	45:det	_
45	grandson	grandson	NOUN	NN	Number=Sing	41	ccomp	41:ccomp	_
46	of	of	ADP	IN	_	50	case	50:case	_
47	a	a	DET	DT	Definite=Ind|PronType=Art	50	det	50:det	_
48	Connecticut	Connecticut	PROPN	NNP	Number=Sing	50	compound	50:compound	_
49	US	US	PROPN	NNP	Number=Sing	50	compound	50:compound	_
50	senator	senator	PROPN	NNP	Number=Sing	45	nmod	45:nmod:of	SpaceAfter=No
51	.	.	PUNCT	.	_	1	punct	1:punct	_
```

File: not-to-release/sources/reviews/063549.xml.conllu [hunk 1]
```diff
@@ -14,7 +14,7 @@
 2	s	be	VERB	VBZ	Mood=Ind|Number=Sing|Person=3|Tense=Pres|VerbForm=Fin	0	root	0:root	_
 3	a	a	DET	DT	Definite=Ind|PronType=Art	4	det	4:det	_
 4	reason	reason	NOUN	NN	Number=Sing	2	nsubj	2:nsubj	_
-5	why	why	SCONJ	WRB	PronType=Int	4	mark	4:mark	_
```
Comment (sebschu) [inline not-to-release/sources/reviews/063549.xml.conllu:17]: LGTM!

Comment (sebschu) [reply not-to-release/sources/reviews/063549.xml.conllu:17]: But `named-9` should be attached to `reason-4`

Comment (amir-zeldes) [reply not-to-release/sources/reviews/063549.xml.conllu:17]: Agreed

**Comment by dan-zeman** [comment]
> weblog-juancole.com_juancole_20040823064025_ENG_20040823_064025-0086
> "Mr. Bush would also sometimes tell stories about his days at Yale in New Haven, and how whenever he got pulled over for erratic driving, he was let go"

> [his days at Yale in New Haven] is coordinated with [how [whenever he got pulled over for erratic driving], he was let go].
> The validator is triggered by the relative clause analysis of "whenever + he got pulled over", which I will fix to match the SCONJ policy. But I am still not sure about the "how": should it remain conj(stories, let), mark(let, how/SCONJ)?

What about `conj(days, let)`? I see two conjoined modifiers of stories: _stories about his days_ and _stories how he was let go_.

**Comment by amir-zeldes** [comment]
Looks good, except for "whenever", where I think I disagree with @sebschu . The "whenever" case was missed due to the double-duty dependency: the rule was basically trying to distinguish WH-complementizers on question adverbials vs. declarative cases comparable to 'if' etc., and in this case there are two SCONJ: 

The "how" is the complementizer of 'let', and governed by 'stories' - stories about how X, similar to "stories about **whether** X", which should be a clear non-adverbial SCONJ.

I see the "whenever" as a complementizer of the "pulled" verb inside an adverbial clause, because it is similar to "**if** he was pulled over", where "if" is again SCONJ, not ADV. @sebschu would you consider the following a free relative? 

`if he got pulled over for erratic driving, he was let go`

If not, I would say:

`mark(let,how), how/SCONJ`
`mark(pulled,whenever), whenever/SCONJ`

**Comment by dan-zeman** [comment]
> But I wonder if it should be considered acl instead, because "reason" is not extracted from a core position of the embedded clause.

I am pretty sure that relative clauses are not restricted to core positions. But I am undecided about whether this particular example should or should not be considered a relative clause. There should be coreference between the relativizer _(why)_ and the modified nominal _(reason)_. It looks a bit like coreference, but a rather vague one.

**Comment by sebschu** [comment]
> I see the "whenever" as a complementizer of the "pulled" verb inside an adverbial clause, because it is similar to "if he was pulled over", where "if" is again SCONJ, not ADV. @sebschu would you consider the following a free relative?

> if he got pulled over for erratic driving, he was let go

No I wouldn't consider this sentece a free relative, but I think this is a paraphrase rather than the same syntactic construction.

I think this sentence has clearly a free relative:

_He was let go whenever he wanted._

I can see how this sentence is semantically different from _Whenever he got pulled over for erratic driving, he was let go_, but I don't see how these two sentences differ syntactically, so I still think this should be a free relative.

**Comment by sebschu** [comment]
> I am pretty sure that relative clauses are not restricted to core positions. 

Yes, definitely. Consider a sentence such as "They sold the house at a loss for which they had originally paid too much". 

But I agree that the coreference in this example is a bit vague.

**Comment by sebschu** [comment]
> What about `conj(days, let)`? 

Yes, this is better! (Given the unlike coordinated phrase, one could also parse the sentence as [stories about ...] and [(stories) how ...] where the second stories was elided but I think that's a bit of a stretch and the analysis @dan-zeman proposed is simpler.)

**Comment by amir-zeldes** [comment]
I think "reason why" is not a relative, which properly should be "the reason due to which ...". For me "reason why" is an adnominal clause, but not relative.

For the free relative question, I think it is the same construction as "if", since I can't spot the point at which the following three diverge syntactically:

  * if he got pulled over for erratic driving, he was let go (everyone agrees this is advcl + mark, right?)
  * when he got pulled over for erratic driving, he was let go (I don't see a difference between if and when here)
  * whenever he got pulled over for erratic driving, he was let go (whenever just clarifies that it's recurring, and not a single instance)

The third case is semantically less ambiguous than the second, since 'when' can refer to a single instance of being pulled over (but doesn't have to). However this semantic difference doesn't make a syntactic difference, at least for me. It's still a conditional adverbial clause specifying what had to apply in order for the 'let go' clause to apply, and syntactically just a CP with the 'wh' word as a C. Is the second clause with 'when' also a free relative? If not, then where is the syntactic difference between 2 and 3?

**Comment by sebschu** [comment]
There may be arguments against one of these two analyses, but I couldn't find any from reading up on free relatives for a bit and maybe it just boils down to preferences in the end.

I find an analysis according to which sentences with _whenever_ and sentences with _wherever_ have the same structure more appealing but I can also see that the parallels to _if_ and _when_ make your analysis more appealing.

And looking at the treebank, it seems like the annotations for sentences with _whenever_ are once again a bit of a mess, so I can't really appeal to conventions either ;)

So generally, I'd be okay with either analysis and ideally we would then apply whichever  analysis we decide upon to all examples in the treebank.

**Comment by nschneid** [comment]
It would be super helpful if somebody could write up guidelines on free relatives and questions/WH-words in UDv2, because I find this area of syntax terribly confusing.

**Comment by nschneid** [comment]
> > 
> 
> > [his days at Yale in New Haven] is coordinated with [how [whenever he got pulled over for erratic driving], he was let go].
> > The validator is triggered by the relative clause analysis of "whenever + he got pulled over", which I will fix to match the SCONJ policy. But I am still not sure about the "how": should it remain conj(stories, let), mark(let, how/SCONJ)?
> 
> What about `conj(days, let)`? I see two conjoined modifiers of stories: _stories about his days_ and _stories how he was let go_.

Oops yes, that is and should be the conj relation.

**Comment by amir-zeldes** [comment]
> It would be super helpful if somebody could write up guidelines on free relatives and questions/WH-words in UDv2, because I find this area of syntax terribly confusing.

I think for me the main criterion is the 'double duty' that free relatives do. For example a sentence like:

`I see what you did`

is problematic for traditional dependencies, because "what" is the object of "see", but also of "did". The free relative analysis treats this like some kind of haplology compared to languages with correlate objects, e.g. German or Polish, which would allow both the English construction, and the more explicit "ich sehe das, was Du gemacht hast" 'I see that, what you did'. In that fuller case, the `acl` nature of the what clause is clearer.

This is not the case for the example we had above IMO, since in:

`whenever he got pulled over, he was let go`

we can analyze 'whenever' in "whenever he got pulled over" as SCONJ (without even looking at the main clause), in which case the UD analysis states that the verb 'pulled' is the head of the advcl, while the SCONJ itself is not construed as an ADV of the main clause. To see the difference, consider the ccomp equivalent of the "what" example:

`He saw you doing what?`

In this case, "doing" is the object of "saw", and "what" is only the object of "doing". For the "whenever" case, we can substitute an adverbial gerund clause without whenever:

`Having been pulled up, he was let go`

This is an indication that 'pulled', and not 'whenever' can be seen as the spatio-temoral adjunct to "let". AFAIK we can't do something like that for the "know what" example. 

BTW I'm not saying "wh*ever" can never be a free relative but I think there's a difference in meaning. Maybe this will clarify what I mean by the double duty criterion:

`Tell me wherever you go`

1. Free relative: inform me which place it is that you're going to (wherever is obj of tell, but also destination of 'go')
2. advcl: no matter where you go, call me from there and tell me (something else). In this case, wherever is the complementizer for the location advcl headed by go, but not the object of tell

**Comment by nschneid** [comment]
Seeing no more suggestions for the PR I'll merge this. :) Ongoing discussion of RCs in #97.

**Comment by nschneid** [comment]
OK now the validator is complaining:

    [L4 Enhanced unknown-edeprel] Unknown enhanced relation type 'advcl:whenever' in '8:advcl:whenever'

@dan-zeman Should this be added to a list of acceptable edeprels or simplified to plain `advcl`?

**Comment by dan-zeman** [comment]
If _whenever_ is now `SCONJ` and `mark` of the adverbial clause then I think it is in accord with the guidelines that the enhanced relation is `advcl:whenever`, and it should be added to [data/edeprel.en](https://github.com/UniversalDependencies/tools/blob/master/data/edeprel.en).

**Comment by nschneid** [comment]
I added it and all hell broke loose with the validator. Does it need to be reloaded or something?

**Comment by dan-zeman** [comment]
It does not seem to be your fault. Roughly at the same time, @rasoolims and @phsfr were playing with the validator source code and managed to push a corrupt version to Github. It seems to be fixed now and the validator is gradually re-validating all the treebanks.

**Comment by nschneid** [comment]
OK, looks like EWT is back to purple. :)

**Pull request merged**